bin/lint_localized_strings_format

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -eu
+
+# Attempts to generate the frozen `.strings` file to gets sent to GlotPress
+# after a new release code freeze.
+#
+# If the process fails, annotates the build with the errors.
+
+# The strings generation happens via Fastlane, so we need to install the
+# Ruby gems.
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+# The strings generation also picks up files from our first party libraries,
+# some of which are installed via CocoaPods
+echo "--- :cocoapods: Setting up Pods"
+install_cocoapods
+
+echo "--- :sleuth_or_spy: Lint Apple Localized Strings Format"
+# A next step improvement is to move the logs management within the release
+# toolkit
+LOGS=logs.txt
+set +e
+set -o pipefail
+if [[ $# -eq 0 ]]; then
+  # If no argument provided, update $@ with default values
+  set -- generate_strings_file_for_glotpress skip_commit:true
+fi
+bundle exec fastlane "$@" | tee $LOGS
+EXIT_CODE=$?
+set -e
+
+if [[ $EXIT_CODE -ne 0 ]]; then
+  # Strings generation finished with errors, extract the errors in an easy-to-find section
+  echo "+++ :x: Strings Generation Failed"
+  ERRORS_FILE=errors.txt
+  printf "Found errors when trying to run \`genstrings\` to generate the \`.strings\` files from \`{NS,App}LocalizedStrings\` calls:\n\n" | tee $ERRORS_FILE
+  # Print the errors inline.
+  #
+  # Notice the second `sed` call that removes the ANSI escape sequences that
+  # Fastlane uses to color the output. We need to do this at this point to
+  # account for Fastlane printing the errors multiple times (inline and in the
+  # end of the lane summary) but with different escape sequences. Without it,
+  # we would get multiple hits for the same error.
+  #
+  # See discussion at
+  # https://github.com/wordpress-mobile/WordPress-iOS/pull/19553#discussion_r1017743950
+  cat $LOGS \
+    | sed -ne 's/\[.*\].*genstrings: error: /- /p' \
+    | sed -e $'s/\x1b\[[0-9;]*m//g' \
```

Comment:
💄  Nitpick: `cat $LOGS | sed -ne $CMD` creates two shell processes (one for `cat` then one for `sed`) so it's usually a better practice to use `sed -ne $CMD $LOGS` to use a single shell process (and let `sed` do the file reading directly, instead of `cat` doing the file IO and `sed` doing stdin-piping) instead 🙃 

I'm actually kinda surprised that `shellcheck` didn't automatically suggest you that (is [it does warn for similar cases of unnecessary uses of `cat`](https://www.shellcheck.net/wiki/SC2002)) 🤷 

```suggestion
  sed -ne 's/\[.*\].*genstrings: error: /- /p' $LOGS \
    | sed -e $'s/\x1b\[[0-9;]*m//g' \
```

(PS: If we want to go even further, I think you could also collapse the to `sed` invocations into a single invocation with a unified sed command/script doing the replacements all at once… but I'm not convinced that will lead to something super readable, so I'm ok to keep those separate)